src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newObj = { ...state };
+  const result = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const type = actions[i].type;
+
+    if (type === 'addProperties') {
+      const extraData = actions[i].extraData;
+
+      for (const key in extraData) {
+        newObj[key] = extraData[key];
+      }
+    } else if (type === 'removeProperties') {
+      const remove = actions[i].keysToRemove;
+
+      for (const key in remove) {
+        delete newObj[remove[key]];
+      }
+      // result.push(Object.assign({}, newObj));
+    } else if (type === 'clear') {
+      Object.keys(newObj).forEach((key) => delete newObj[key]);
+    }
+    result.push(Object.assign({}, newObj));
```

Comment:
For future, you can re-write it like this or use a switch case
```suggestion
    if (type === 'addProperties') {
      const extraData = actions[i].extraData;

      for (const key in extraData) {
        newObj[key] = extraData[key];
      }
    } 
    
    if (type === 'removeProperties') {
      const remove = actions[i].keysToRemove;

      for (const key in remove) {
        delete newObj[remove[key]];
      }
      // result.push(Object.assign({}, newObj));
    } 
    
    if (type === 'clear') {
      Object.keys(newObj).forEach((key) => delete newObj[key]);
    }
    
    result.push(Object.assign({}, newObj));
```